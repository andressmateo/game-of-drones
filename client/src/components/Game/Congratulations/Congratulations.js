import React from 'react';
import styled from 'styled-components';

import { Card, Button } from '../../StyledComponents';

const Container = Card.extend`
  width: 500px;
  height: 260px;
  padding-top: 80px;
`;

const Congratulations = ({ name, playAgain }) => {
  return (
    <Container>
      <h1>We have a WINNER!!</h1>
      <h2>{name} is the new EMPEROR!</h2>
      <br />
      <Button onClick={() => playAgain()}>Play Again</Button>
    </Container>
  );
};

export default Congratulations;
