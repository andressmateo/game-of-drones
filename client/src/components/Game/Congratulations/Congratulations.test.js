import React from 'react';
import { shallow } from 'enzyme';
import chai, { expect } from 'chai';
import renderer from 'react-test-renderer';
import chaiEnzyme from 'chai-enzyme';
import chaiJestSnapshot from 'chai-jest-snapshot';
import sinon from 'sinon';

import Congratulations from './Congratulations';

chai.use(chaiEnzyme());
chai.use(chaiJestSnapshot);

describe('Congratulations component', () => {
  it('renders without crashing', () => {
    shallow(<Congratulations />);
  });

  it('renders correctly - snapshot', () => {
    const tree = renderer.create(<Congratulations />);
    expect(tree).to.matchSnapshot();
  });

  it('Calls `play again` when is requested', () => {
    const playAgainSpy = sinon.spy();
    const wrapper = shallow(<Congratulations playAgain={playAgainSpy} />);
    wrapper.find('#play-again').simulate('click');
    expect(playAgainSpy.calledOnce).to.be.true;
  });
});
