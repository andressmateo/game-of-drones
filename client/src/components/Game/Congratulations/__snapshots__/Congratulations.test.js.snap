// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Congratulations component renders correctly - snapshot 1`] = `
<div
  className="sc-bxivhb khVRMm"
>
  <h1>
    We have a WINNER!!
  </h1>
  <h2>
     is the new EMPEROR!
  </h2>
  <br />
  <button
    className="sc-bwzfXH fMRgfY"
    id="play-again"
    onClick={[Function]}
  >
    Play Again
  </button>
</div>
`;
